runtime/vm/ContinuationHelpers.cpp

```diff
@@ -0,0 +1,167 @@
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+#include "j9.h"
+#include "j9comp.h"
+#include "j9protos.h"
+#include "ut_j9vm.h"
+#include "vm_api.h"
+#include "OutOfLineINL.hpp"
+#include "ContinuationHelpers.hpp"
+
+extern "C" {
+#if JAVA_SPEC_VERSION >= 19
+
+J9_DECLARE_CONSTANT_UTF8(continuationClass_name, "jdk/internal/vm/Continuation");
+J9_DECLARE_CONSTANT_UTF8(execute_sig, "(Ljdk/internal/vm/Continuation;)V");
+J9_DECLARE_CONSTANT_UTF8(execute_name, "execute");
+
+extern void c_cInterpreter(J9VMThread *currentThread);
+
+BOOLEAN
+createContinuation(J9VMThread *currentThread, j9object_t continuationObject)
+{
+	J9JavaVM *vm = currentThread->javaVM;
+	PORT_ACCESS_FROM_PORT(vm->portLibrary);
+	BOOLEAN result = TRUE;
+	J9JavaStack *stack = NULL;
+	J9SFJNINativeMethodFrame *frame = NULL;
+
+	J9VMContinuation *continuation = (J9VMContinuation *)j9mem_allocate_memory(sizeof(J9VMContinuation), OMRMEM_CATEGORY_THREADS);
+	if (NULL == continuation) {
+		vm->internalVMFunctions->setNativeOutOfMemoryError(currentThread, 0, 0);
+		result = FALSE;
+		goto end;
+	}
+
+	memset(continuation, 0, sizeof(J9VMContinuation));
+
+#ifdef J9VM_INTERP_GROWABLE_STACKS
+#define VMTHR_INITIAL_STACK_SIZE ((vm->initialStackSize > (UDATA) vm->stackSize) ? vm->stackSize : vm->initialStackSize)
+#else
+#define VMTHR_INITIAL_STACK_SIZE vm->stackSize
+#endif
+
+	if ((stack = allocateJavaStack(vm, VMTHR_INITIAL_STACK_SIZE, NULL)) == NULL) {
+		vm->internalVMFunctions->setNativeOutOfMemoryError(currentThread, 0, 0);
+		result = FALSE;
+		goto end;
+	}
+
+#undef VMTHR_INITIAL_STACK_SIZE
+
+	continuation->stackObject = stack;
+	continuation->stackOverflowMark = continuation->stackOverflowMark2 = J9JAVASTACK_STACKOVERFLOWMARK(stack);
+
+	frame = ((J9SFJNINativeMethodFrame*)stack->end) - 1;
+	frame->method = NULL;
+	frame->specialFrameFlags = 0;
+	frame->savedCP = NULL;
+	frame->savedPC = (U_8*)(UDATA)J9SF_FRAME_TYPE_END_OF_STACK;
+	frame->savedA0 = (UDATA*)(UDATA)J9SF_A0_INVISIBLE_TAG;
+	continuation->sp = (UDATA*)frame;
+	continuation->literals = (J9Method*)0;
+	continuation->pc = (U_8*)J9SF_FRAME_TYPE_JNI_NATIVE_METHOD;
+	continuation->arg0EA = (UDATA*)&frame->savedA0;
+	continuation->stackObject->isVirtual = TRUE;
```

Comment:
need to set this to `FALSE` on the carrier/platform  `allocateJavaStack(..)` case

Reply:
This call would only be used to create continuation stack, not sure how it would be used by carrier/platform

Reply:
What I mean is that, we've added a field to `J9JavaStack` we need to make sure its properly initialized in the non-VirtualThread case